src/setup_payload/AdditionalDataPayloadGenerator.cpp

```diff
@@ -51,35 +51,34 @@ AdditionalDataPayloadGenerator::generateAdditionalDataPayload(uint16_t lifetimeC
 {
     System::PacketBufferTLVWriter writer;
     TLVWriter innerWriter;
-    char rotatingDeviceIdBuffer[RotatingDeviceId::kHexMaxLength];
+    uint8_t rotatingDeviceIdBuffer[RotatingDeviceId::kHexMaxLength];
```

Comment:
That does not look like the right length for the _binary_ version of rotating device id.
```suggestion
    uint8_t rotatingDeviceIdBuffer[RotatingDeviceId::kMaxLength];
```